workshop/content/lab1.5_kafka.md

```diff
@@ -0,0 +1,272 @@
+imc-dispatcher-575c7fcd8d-lrpmt                    1/1     Running     0          3d13h
+mt-broker-controller-56857cccc5-h49sp              1/1     Running     0          3d13h
+mt-broker-filter-784b7db965-5ngkk                  1/1     Running     0          3d13h
+mt-broker-ingress-6b9f847866-bhk5w                 1/1     Running     0          3d13h
+sugar-controller-594784974b-rpvsm                  1/1     Running     0          3d13h
+```
+
+We can also see the various kinds of objects we will be using, starting first with the eventing sources.  Notice the `KafkaSource`.
+
+```
+oc api-resources --api-group='sources.knative.dev'
+```
+
+```
+NAME               SHORTNAMES   APIGROUP              NAMESPACED   KIND
+apiserversources                sources.knative.dev   true         ApiServerSource
+containersources                sources.knative.dev   true         ContainerSource
+kafkasources                    sources.knative.dev   true         KafkaSource
+pingsources                     sources.knative.dev   true         PingSource
+sinkbindings                    sources.knative.dev   true         SinkBinding
+```
+
+Next we can see the specific eventing/messaging resources.  Notice the `Channel`.
+
+```
+oc api-resources --api-group='messaging.knative.dev'
+```
+
+```
+NAME               SHORTNAMES   APIGROUP                NAMESPACED   KIND
+channels           ch           messaging.knative.dev   true         Channel
+inmemorychannels   imc          messaging.knative.dev   true         InMemoryChannel
+subscriptions      sub          messaging.knative.dev   true         Subscription
+```
+
+### Knative Channel
+Channels are the mechanism that actually forward events through the system, from the source (producer) to the sink (consumer). By default, all Knative Eventing API use the InMemoryChannel (imc), which is great to get started, but is not applicable for all use cases due to it's inability to persist messages. To get durability, we need to use other channels, such as GCP PubSub or Kafka.
+
+
+### Verify Channel
+Let's verify that we are configured to use the `KafkaChannel`.
+
+```
+oc get configmap config-default-kafka-channel -o yaml -n knative-eventing
+```
+
+You can see the default for `kafka` is `KafkaChannel`.  Here's the relevant snippet:
+
+```
+apiVersion: v1
+data:
+  default-ch-config: |
+    namespaceDefaults:
+      kafka:
+        apiVersion: messaging.knative.dev/v1alpha1
+        kind: KafkaChannel
+```
+
+### Create Sink
+We will now create a sink, a final destination for Kafka events flowing through the system. In this case, it will be a Serverless service. Let's create one now.
+
+First let's make sure you are still in the right project.
+
+```
+oc project
+# Using project "user-$USER_NAME" on server...
+```
+
+Now we can create our sink.
+
+```
+oc apply -f https://raw.githubusercontent.com/RedHatGov/serverless-workshop-code/main/kafka/kafka-sink.yml
+```
+
+Verify that it is there:
```

Comment:
Verify that it is there and wait until it is ready:

Reply:
👍